### PR TITLE
[Docs] Add gtk-doc comments for hb-ot-var

### PR DESCRIPTION
**Pull request by n8willis**
This adds GTK-Doc comments and annotations for hb-ot-var.cc and .h.

### CODE REVIEW

File: src/hb-ot-var.cc
Comment (khaledhosny) [inline src/hb-ot-var.cc]:
```suggestion
 * Tests whether a face includes any OpenType variation data in the `fvar` table.
```

File: src/hb-ot-var.cc
Comment (khaledhosny) [inline src/hb-ot-var.cc]: Shouldn’t it be `(out caller-allocates)`?

File: src/hb-ot-var.cc
Comment (khaledhosny) [inline src/hb-ot-var.cc]: I *think* `#hb_ot_var_get_axis_infos` should be `hb_ot_var_get_axis_infos()` since the in GTK-Doc `#` prefix is for types is for types, while functions are detected by the presence of `()` suffix.

File: src/hb-ot-var.cc
Comment (khaledhosny) [inline src/hb-ot-var.cc]: `(out caller-allocates)`?

File: src/hb-ot-var.cc
Comment (khaledhosny) [inline src/hb-ot-var.cc]:
```suggestion
 * Fetches the `name` table Name ID that provides display names for
```

File: src/hb-ot-var.cc
Comment (khaledhosny) [inline src/hb-ot-var.cc]:
```suggestion
 * Fetches the `name` table Name ID that provides display names for
```

File: src/hb-ot-var.cc
Comment (khaledhosny) [inline src/hb-ot-var.cc]:
```suggestion
 * Deprecated: 2.2.0: use hb_ot_var_get_axis_infos() instead
```

Based on the example from https://developer.gnome.org/gtk-doc-manual/stable/documenting_symbols.html.en

File: src/hb-ot-var.cc
Comment (n8willis) [reply src/hb-ot-var.cc]: Honestly: I'm not sure. You have the experience with GObject Introspection etc that I don't, so I trust your judgement on it.

File: src/hb-ot-var.cc
Comment (n8willis) [reply src/hb-ot-var.cc]: OK. I thought # was more general than that, but upon searching I do see some instances of () getting correctly formatted. I'll change here, then at some point after the merging it's worth me doing a separate consistency pass on all the comments new & old.

File: src/hb-ot-var.cc
Comment (khaledhosny) [reply src/hb-ot-var.cc]: May memory is fading, so whatever is done elsewhere and lets see if anyone complains.

Comment (n8willis) [reply src/hb-ot-var.cc]: @behdad Any quick answer there?

**Comment by khaledhosny** [comment]
@behdad I reviewed this as well, I think @n8willis is waiting for an answer from you to his  question above.

File: src/hb-ot-var.cc
Comment (behdad) [reply src/hb-ot-var.cc]: > Shouldn’t it be `(out caller-allocates)`?

That's what we do in other places it seems, yes.  Though, by far not in all places.  I think it should be added to many more places.

Comment (behdad) [reply src/hb-ot-var.cc]: Probably all of these need fixing:
```
$ git grep '(out).*(array length='
hb-buffer-serialize.cc: * @buf: (out) (array length=buf_size) (element-type uint8_t): output string to
hb-ot-color.cc: * @colors: (out) (array length=color_count) (nullable): The array of #hb_color_t records found
hb-ot-color.cc: * @layers: (out) (array length=layer_count) (nullable): The array of layers found
hb-ot-layout.cc: * @point_array: (out) (array length=point_count): The array of attachment points found for the query
hb-ot-layout.cc: * @caret_array: (out) (array length=caret_count): The array of caret positions found for the query
hb-ot-layout.cc: * @script_tags: (out) (array length=script_count): The array of #hb_tag_t script tags found for the query
hb-ot-layout.cc: * @feature_tags: (out) (array length=feature_count): Array of feature tags found in the table
hb-ot-layout.cc: * @language_tags: (out) (array length=language_count): Array of language tags found in the table
hb-ot-layout.cc: * @feature_indexes: (out) (array length=feature_count): The array of feature indexes found for the query
hb-ot-layout.cc: * @feature_tags: (out) (array length=feature_count): The array of #hb_tag_t feature tags found for the query
hb-ot-layout.cc: * @lookup_indexes: (out) (array length=lookup_count): The array of lookup indexes found for the query
hb-ot-layout.cc: * @lookup_indexes: (out) (array length=lookup_count): The array of lookups found for the query
hb-ot-math.cc: * @variants: (out) (array length=variants_count): array of variants returned
hb-ot-math.cc: * @parts: (out) (array length=parts_count): the glyph parts returned
hb-ot-name.cc: * Returns: (out) (transfer none) (array length=num_entries): Array of available name entries.
```
except for the last one I think.

**Review by khaledhosny — APPROVED** [review]

**Comment by khaledhosny** [comment]
I applied a couple of minor fixes and rebased.

**Review by alerque — APPROVED** [review]

**Pull request merged**
